modules/highgui/src/window_w32.cpp

```diff
@@ -1874,21 +1874,41 @@ static void showSaveDialog(CvWindow* window)
     ofn.lStructSize = sizeof(ofn);
 #endif
     ofn.hwndOwner = window->hwnd;
+#ifdef HAVE_PNG   
     ofn.lpstrFilter = "Portable Network Graphics files (*.png)\0*.png\0"
+#else
+    ofn.lpstrFilter = "Windows bitmap (*.bmp;*.dib)\0*.bmp;*.dib\0"
+#endif
+#ifdef HAVE_JPEG
                       "JPEG files (*.jpeg;*.jpg;*.jpe)\0*.jpeg;*.jpg;*.jpe\0"
+#endif
+#ifdef HAVE_PNG 
                       "Windows bitmap (*.bmp;*.dib)\0*.bmp;*.dib\0"
```

Comment:
Looks like it is always supported, because it is implemented in OpenCV directly without any `ifdef`:
https://github.com/Itseez/opencv/blob/0726c4d4ea80e73c96ccee7bd3ef5f71f46ac82b/modules/imgcodecs/src/grfmt_hdr.cpp#L118